scripts/release_scripts/get_repo_specific_changes.py

```diff
@@ -0,0 +1,159 @@
+            VERSION_RE_FORMAT_STRING % variable, old_feconf)[0]
+        new_version = re.findall(
+            VERSION_RE_FORMAT_STRING % variable, new_feconf)[0]
+        if old_version != new_version:
+            changed_version_vars_in_feconf.append(variable)
+    return changed_version_vars_in_feconf
+
+
+def _git_diff_names_only(first_release_tag, second_release_tag='HEAD'):
+    """Get names of changed files from git between two releases.
+
+    Args:
+        first_release_tag: str. The first release tag to diff against.
+        second_release_tag: str. The second release tag to diff against.
+
+    Returns:
+        list(str). List of filenames that are different between two releases.
+    """
+    diff_cmd = (
+        GIT_CMD_DIFF_NAMES_ONLY_FORMAT_STRING % (
+            first_release_tag, second_release_tag))
+    return common.run_cmd(diff_cmd.split(' ')).splitlines()
+
+
+def get_changes_in_setup_scripts(
+        release_tag_to_diff_against, only_include_changed_scripts=True):
+    """Returns a dict of setup script with a status of whether they have
+    changed or not since the release against which diff is being checked.
+
+    Args:
+        release_tag_to_diff_against: str. The release tag to diff against.
+        only_include_changed_scripts: bool. If true the output dict only
+            include the setup scripts which changed. If false, the output dict
+            will include the status for all setup scripts which are checked
+            irrespective of whether they are changed or not.
+
+    Returns:
+        dict. Dict consisting of key as script name and value as boolean
+        indicating whether or not it has changed
+        (filtered by default to those that are modified).
+    """
+    setup_scripts = ['scripts/%s' % item for item in
+                     ['setup.py', 'setup_gae.py', 'install_third_party_libs.py',
+                      'install_third_party.py']]
+    changed_files = _git_diff_names_only(release_tag_to_diff_against)
+    changes_dict = {script: script in changed_files
+                    for script in setup_scripts}
+    if only_include_changed_scripts:
+        return {name: status for name, status
+                in changes_dict.items() if status}
+    else:
+        return changes_dict
+
+
+def get_changes_in_storage_models(release_tag_to_diff_against):
+    """Returns a list of filepaths in core/storage whose contents have
+    changed since the release against which diff is being checked.
+
+    Args:
+        release_tag_to_diff_against: str. The release tag to diff against.
+
+    Returns:
+        list(str). The changed filenames in core/storage (if any).
+    """
+    diff_list = _git_diff_names_only(release_tag_to_diff_against)
+    return [item for item in diff_list if item.startswith('core/storage')]
+
+
+def get_changes(release_tag_to_diff_against):
+    """Collects changes in storage models, setup scripts and feconf
+    since the specified release in args.
```

Comment:
What do you mean by "since the specified release in args"?

Reply:
Updated.